src/extension.ts

```diff
@@ -176,4 +193,16 @@ export function activate(context: vscode.ExtensionContext): void {
             !shouldDisableErrorMessage() && showContactUsDetails(context.logUri, logFileName);
         }
     }, 300, {});
+
+    const handleScanResults = (filename: string, editor: vscode.TextEditor, state: vscode.Memento, checkovFails: FailedCheckovCheck[]) => {
+        saveCheckovResult(context.workspaceState, checkovFails);
+        applyDiagnostics(editor.document, diagnostics, checkovFails);
+        (checkovFails.length > 0 ? setErrorStatusBarItem : setPassedStatusBarItem)(checkovInstallation?.version);
+        const hash = getFileHash(filename);
+        fileCache.set(hash, {
```

Comment:
I chose the hash because that way it works with scanning a temporary file (when we scan a file that has been edited, but not saved). In my testing, this made it feel much smoother and intuitive than my first version, which did use the filename as the key.